### DISCUSSION TITLE
GitHub / PnP Search code security / assurance

### DISCUSSION THREAD

**Discussion by muira** [microsoft-search/pnp-modern-search#2299] [Q&A]
Hi all, I have a non technical query which could be generalised to a broader git hub post, but putting here as I'm really only after the PnP Search web parts, which are fantastic, thank you!

If i want to deploy to our tenant I'm going to have to present some reassurance to our Technical Design Authority around the security / safety of deploying a community built solution.

I read that there are optional code scanning tool available on GitHub for contributors..

I read this line from the docs:

"While allowing flexibility it introduces security measures to block JavaScript and CSS injection, key to many of the enterprise companies using the web parts today in productions."

And i have it that Microsoft recommend this solution, which is a good start, but is there anything else i can pass on that might placate the questions around code security..?

Thank you

Andy

**Comment by FranckyC** [comment]
Hi @muira, maybe this can help: https://github.com/microsoft-search/pnp-modern-search/discussions/2139

**Comment by muira** [comment]
Hi Franck, it's another piece of the puzzle, i'll collate everything i have and hopefully it's enough!

thank you.

**Reply by wobba** [reply]
As far as we know we have addressed all security issues known., but no guarantee that there are not others.

So it's about trusting the code, or reviewing it. In terms of libs with vulnerabilities we do update these when these issues are alerted. And yes, the solution is used widespread, but I cannot share numbers or companies.

Hosting the solution on a Microsoft owned repo might give more trust to someone asking.

I'd say, if you trust a consultant to write SPFx for you, you should be able to trust this solution.

**Reply by muira** [reply]
Thank you Wobba, of course you're absolutely right. I'm fully confident that there's no risk here, i'll go have a chat with some people and hopefully get this green lit. thank you!

**Reply by wobba** [reply]
@muira best we can do :) Either you trust the open source solution you are using, or you have someone front that solution on your behalf, which means if there's an error the vendor/consultant installing it takes responsibility to have issues fixed/resolved.

**Reply by muira** [reply]
Hi Wobba, and that person is... me! yes that's fine, I'm happy to stand up and say i trust this stuff implicitly, I think it's clear that it's all very well managed. Thank you
